src/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,175 @@
+    status = "Status",
+    message = "Message",
+    name = "Data.Name",
+    value = "Data.Value",
+    expanded = "Expanded",
+    solutions = "Solutions",
+    time = "Time",
+    instance = "Instance",
+    facility = "Facility",
+    failureCount = "FailureCount"
+}
+
+export const detectorCancelTokens: Map<string, CancellationTokenSource> = new Map();
+export async function checkLinuxWebAppDownDetector(node: SiteTreeItem, tokenSource: CancellationTokenSource): Promise<void> {
+    return await callWithTelemetryAndErrorHandling('appService.linuxWebAppDownDetector', async (context: IActionContext): Promise<void> => {
+        const deployment: DeployResult = await node.root.client.kudu.deployment.getResult('latest');
+        if (!deployment.startTime) {
+            // if there's no deployment detected, nothing can be done
+            context.telemetry.properties.cancelStep = 'noDeployResult';
+            return;
+        }
+
+        const deployResultTime: Date = new Date(deployment.startTime);
+
+        const detectorOutput: string = `Diagnosing web app "${node.root.client.siteName}" for critical errors...`;
+        ext.outputChannel.appendLine(detectorOutput);
+
+        const detectorUri: string = `${node.root.environment.resourceManagerEndpointUrl}${node.id}/detectors/${detectorId}`;
+        const requestOptions: WebResource & Partial<{ qs: queryString }> = new WebResource();
+
+        requestOptions.method = 'GET';
+        requestOptions.url = detectorUri;
+        requestOptions.qs = {
+            'api-version': "2015-08-01",
+            fId: "1",
+            btnId: "2",
+            inpId: "1",
+            val: "vscode",
+            startTime: deployResultTime.toISOString(),
+            endTime: new Date().toISOString()
+        };
+        await signRequest(requestOptions, node.root.credentials);
+
+        let detectorErrorMessage: string | undefined;
+        // wait 10 minutes for a response from the detector
+        const detectorTimeoutMs: number = Date.now() + 20 * 60 * 1000;
+        do {
+            if (tokenSource.token.isCancellationRequested) {
+                // the user cancelled the check by deploying again
+                throw new UserCancelledError();
+            }
+
+            if (Date.now() > detectorTimeoutMs) {
+                const noIssuesFound: string = `No critical issues found for web app "${node.root.client.siteName}".`;
+                ext.outputChannel.appendLine(noIssuesFound);
+                return undefined;
+            }
+            // update the new end time to be now
+            requestOptions.qs.endTime = new Date().toISOString();
+            detectorErrorMessage = await getLinuxDetectorError(requestOptions, deployResultTime, node.root.client.fullName);
+            // poll every minute
+            await delay(1000 * 60);
+        } while (!detectorErrorMessage);
+
+        await ext.ui.showWarningMessage(detectorErrorMessage, { title: 'Open in Portal' });
```

Comment:
Per offline discussion, we're gonna either hide this message for now or add a feature flag. At least until more people on our team (especially Matt) can try it out and see how they feel about the experience